src/sentry.ts

```diff
@@ -0,0 +1,181 @@
+        } catch (e) {}
+    }
+    if (navigator.storage && navigator.storage.estimate) {
+        try {
+            const estimate = await navigator.storage.estimate();
+            result["storageManager_quota"] = String(estimate.quota);
+            result["storageManager_usage"] = String(estimate.usage);
+            if (estimate.usageDetails) {
+                Object.keys(estimate.usageDetails).forEach(k => {
+                    result[`storageManager_usage_${k}`] = String(estimate.usageDetails[k]);
+                });
+            }
+        } catch (e) {}
+    }
+
+    return result;
+}
+
+function getUserContext(client: MatrixClient): Record<string, string> {
+    return {
+        "username": client.credentials.userId,
+        "enabled_labs": getEnabledLabs(),
+        "low_bandwidth": SettingsStore.getValue("lowBandwidth") ? "enabled" : "disabled",
+    };
+}
+
+function getEnabledLabs(): string {
+    const enabledLabs = SettingsStore.getFeatureSettingNames().filter(f => SettingsStore.getValue(f));
+    if (enabledLabs.length) {
+        return enabledLabs.join(", ");
+    }
+}
+
+async function getCryptoContext(client: MatrixClient): Promise<Record<string, string>> {
+    if (!client.isCryptoEnabled()) {
+        return {};
+    }
+    const keys = [`ed25519:${client.getDeviceEd25519Key()}`];
+    if (client.getDeviceCurve25519Key) {
+        keys.push(`curve25519:${client.getDeviceCurve25519Key()}`);
+    }
+    const crossSigning = client.crypto.crossSigningInfo;
+    const secretStorage = client.crypto.secretStorage;
+    const pkCache = client.getCrossSigningCacheCallbacks();
+    const sessionBackupKeyFromCache = await client.crypto.getSessionBackupPrivateKey();
+
+    return {
+        "device_keys": keys.join(', '),
+        "cross_signing_ready": String(await client.isCrossSigningReady()),
+        "cross_signing_supported_by_hs":
+            String(await client.doesServerSupportUnstableFeature("org.matrix.e2e_cross_signing")),
+        "cross_signing_key": crossSigning.getId(),
+        "cross_signing_privkey_in_secret_storage": String(
+            !!(await crossSigning.isStoredInSecretStorage(secretStorage))),
+        "cross_signing_master_privkey_cached": String(
+            !!(pkCache && await pkCache.getCrossSigningKeyCache("master"))),
+        "cross_signing_user_signing_privkey_cached": String(
+            !!(pkCache && await pkCache.getCrossSigningKeyCache("user_signing"))),
+        "secret_storage_ready": String(await client.isSecretStorageReady()),
+        "secret_storage_key_in_account": String(!!(await secretStorage.hasKey())),
+        "session_backup_key_in_secret_storage": String(!!(await client.isKeyBackupKeyStored())),
+        "session_backup_key_cached": String(!!sessionBackupKeyFromCache),
+        "session_backup_key_well_formed": String(sessionBackupKeyFromCache instanceof Uint8Array),
+    };
+}
+
+function getDeviceContext(client: MatrixClient): Record<string, string> {
```

Comment:
Can you type it somehow, so it won't be a `Record<string, string>` but at least a `Record<string, keyof SOMETHING>`? Especially that there's so many keys in here that can or can not exist, adding a proper type here would be very useful.